### PR TITLE
refactor: Make `HasSpecs` implementation more functional

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Make `specs` and `pypi_deps` more functional

### CODE REVIEW

File: src/cli/has_specs.rs [hunk 1]
```diff
@@ -12,27 +12,30 @@ pub(crate) trait HasSpecs {
...
-                miette::miette!("could not find package name in MatchSpec {}", spec)
-            })?;
-            map.insert(name, spec);
-        }
-        Ok(map)
+        self.packages()
+            .iter()
+            .map(|package| {
+                let spec =
+                    MatchSpec::from_str(package, ParseStrictness::Strict).into_diagnostic()?;
```
Comment (baszalmstra) [inline src/cli/has_specs.rs:19]:
```suggestion
                    MatchSpec::from_str(package, ParseStrictness::Lenient).into_diagnostic()?;
```

**Review by baszalmstra — CHANGES_REQUESTED** [review]

File: src/cli/has_specs.rs (hunk 1 above)
Comment (Hofer-Julian) [reply src/cli/has_specs.rs:19]: I will trust you on that :)

Comment (baszalmstra) [reply src/cli/has_specs.rs]: I had just fixed that in #1852

**Review by baszalmstra — APPROVED** [review]

File: src/cli/has_specs.rs (hunk 1 above)
Comment (Hofer-Julian) [reply src/cli/has_specs.rs]: Ah. That would have been funny if i had immediately reverted that. Nice find 😁

**Pull request merged**